src/components/users/UserLoginForm.js

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react'
+import emailValidator from 'email-validator'
+
+import RaisedButton from 'material-ui/RaisedButton'
+
+import TextField from './TextFieldExtended'
+
+class UserLoginForm extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      password: null,
+      email: null,
+
+      validFields: {
+        password: false,
+        email: false
+      }
+    }
+  }
+
+  onFieldValueChange = (fieldName, fieldValue, hasError) => {
+    let newState = Object.assign({}, this.state)
+    newState.validFields[fieldName] = !hasError
+    newState[fieldName] = fieldValue
+    this.setState(newState)
```

Comment:
If you want to use something from previous state, always use updater function:

```
this.setState((prevState, props) => {
  return {counter: prevState.counter + props.step};
});
```

Read more about it here: https://reactjs.org/docs/react-component.html#setstate